backend/src/api/events.js

```diff
@@ -0,0 +1,196 @@
+    let eventDescription = data.description
+
+    if (eventName == undefined) {
+      response = 'Invalid Event Name'
+    } else if (eventDate == undefined) {
+      response = 'Invalid Event Date'
+    } else if (eventStartTime == undefined) {
+      response = 'Invalid Start Time'
+    } else if (eventEndTime == undefined) {
+      response = 'Invalid End Time'
+    } else if (eventLocation == undefined) {
+      response = 'Invalid Location'
+    } else if (eventDescription == undefined) {
+      response = 'Invalid Description'
+    } else {
+      const newEvent = new Event({
+        name: eventName,
+        date: eventDate,
+        startTime: eventStartTime,
+        endTime: eventEndTime,
+        location: eventLocation,
+        description: eventDescription,
+        attendees: []
+      })
+      await newEvent.save()
+      code = 200
+      success = true
+    }
+    res.json({
+      code,
+      message: response,
+      result: {},
+      success
+    })
+  })
+)
+
+// update an event
+router.put(
+  '/:event_id',
+  errorWrap(async (req, res) => {
+    const data = req.body
+    let response = 'Event Edited Sucessfully'
+    let eventId = req.params.event_id
+    let newName = data.name
+    let newDate = data.date
+    let newStartTime = data.startTime
+    let newEndTime = data.endTime
+    let newLocation = data.location
+    let newDescription = data.description
+
+    if (eventId == undefined) {
+      response = 'Invalid Edit Event Request'
+    } else {
+      if (newName != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { name: newName } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newDate != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { date: newDate } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newStartTime != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { startTime: newStartTime } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newEndTime != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { endTime: newEndTime } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newLocation != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { location: newLocation } },
+          { new: true },
+          function(err, doc) {}
+        )
+      }
+      if (newDescription != undefined) {
+        Event.findOneAndUpdate(
+          { _id: new mongodb.ObjectId(eventId) },
+          { $set: { description: newDescription } },
+          { new: true },
+          function(err, doc) {}
```

Comment:
Let's try to avoid a bunch of separate database calls. We can probably do this only using one interaction with the database.

Reply:
How would you consolidate into 1 database call? I was just going off of the code in candidates.js and interview.js lol

Reply:
Yeah I figured lol, it does look like this in other places too, but we might as well change it now rather than increasing our tech debt! So with `$set` you should just be able to pass multiple key value pairs in json object (`{a: val1, b: val2}`). Maybe use your conditional statements to set keys to values in your json, if they are not undefined. And then after all the conditionals just issue one mongo request. Ideally, we could make it cleaner (not have `n` conditionals for `n` attributes) but that's not super necessary as long as we can just do 1 mongo request here.